server/src/main/java/io/druid/segment/realtime/appenderator/Appenderators.java

```diff
@@ -124,8 +124,8 @@ public boolean isAnnounced(DataSegment segment)
         null,
         indexIO,
         indexMerger,
-        null,
-        null
+        MapCache.create(500000),
```

Comment:
Doesn't seem to be universal to use specifically `MapCache` (`CaffeineCache` is also an in-memory, local cache) of the specific size. Maybe replace `Appendirators` with something like `AppendiratorBuilder`, which allows to configure those batteries of arguments one-by-one using named configuration methods? It may distinguish between "realtime" and "offline" on different stages, e. g. in the beginning of the chain:
```java
AppenderatorBuilder.offline() // or .realtime(segmentAnnouncer)
  .setIndexMerger(...)
  .setCache(...)
  .build();
```

or in the end:
```java
AppenderatorBuilder.create()
  .setIndexMerger(..)
  .setCache(..)
  .buildRealtime() // or .buildOffline()
```

Or in some other way